autoscale-core/src/main/java/com/hpe/caf/autoscale/core/GovernorImpl.java

```diff
@@ -178,4 +227,72 @@ private boolean shouldBeScaledDown(final int currentMemoryLoadLimit, final int s
             }
         }
     }
+
+    private static boolean isStringNullOrEmpty(final String string)
+    {
+        return string == null || string.isEmpty();
+    }
+
+    private final class AdvancedInstanceInfo
+    {
+        private final InstanceInfo instanceInfo;
```

Comment:
Looking through the accessor methods it would appear that inheriting from `InstanceInfo` might be a better choice, as it would avoid the need for most of them.  Is that not the case?

Reply:
I wouldn't have expected this instance variable to remain after the move to inheritance.